### ISSUE TITLE
cilium config --all prints masks wrong

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#18106]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

```
# cilium config --all | grep -B 2 Mask
LoadBalancerRSSv4
        IP                        : 
        Mask                      : <nil>
--
LoadBalancerRSSv6
        IP                        : 
        Mask                      : <nil>
--
NAT46Prefix
        IP                        : 0.0.0.0
        Mask                      : ////////////////AAAAAA==
```

`Mask` is formatted as random bytes output rather than as a human-readable mask such as `255.255.255.0` or `/24`

### Cilium Version

```
# cilium version
Client: 1.11.0-rc3 c9d9f7e 2021-11-23T01:47:39+01:00 go version go1.17.3 linux/amd64
Daemon: 1.11.0-rc3 c9d9f7e 2021-11-23T01:47:39+01:00 go version go1.17.3 linux/amd64
```

### Kernel Version

Any

### Kubernetes Version

Any

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sayboras** [comment]
related to https://github.com/golang/go/issues/40436

**Comment by Nagendra-boyina** [comment]
I would like to take up this issue , could you please assign to me?

**Comment by dipankardas011** [comment]
is this issue still relevant
if yes can I work on this

just can you summarize any changes to this issue or what is the objective?

**Comment by dipankardas011** [comment]
> ar as I know this problem is not yet s

Thanks for the head up!, will check and discuss with you :)

**Comment by dipankardas011** [comment]
@joestringer Unable to run this command

Error message
```bash
$ cilium config view --all | grep -B 2 Mask
unknown flag: --all
```

then tried
```bash
cilium config view
agent-not-ready-taint-key              node.cilium.io/agent-not-ready
annotate-k8s-node                      true
arping-refresh-period                  30s
auto-direct-node-routes                false
bpf-lb-external-clusterip              false
bpf-lb-map-max                         65536
bpf-map-dynamic-size-ratio             0.0025
bpf-policy-map-max                     16384
cgroup-root                            /sys/fs/cgroup
cilium-endpoint-gc-interval            5m0s
cluster-id
cluster-name                           default
cluster-pool-ipv4-cidr                 10.0.0.0/8
cluster-pool-ipv4-mask-size            24
custom-cni-conf                        false
debug                                  false
disable-cnp-status-updates             true
disable-endpoint-crd                   false
enable-auto-protect-node-port-range    true
enable-bandwidth-manager               false
enable-bpf-clock-probe                 true
enable-endpoint-health-checking        true
enable-health-check-nodeport           true
enable-health-checking                 true
enable-host-legacy-routing             true
enable-hubble                          true
enable-ipv4                            true
enable-ipv4-masquerade                 true
enable-ipv6                            false
enable-ipv6-masquerade                 true
enable-k8s-terminating-endpoint        true
enable-l2-neigh-discovery              true
enable-l7-proxy                        true
enable-local-redirect-policy           false
enable-policy                          default
enable-remote-node-identity            true
enable-session-affinity                true
enable-well-known-identities           false
enable-xt-socket-fallback              true
hubble-disable-tls                     false
hubble-listen-address                  :4244
hubble-socket-path                     /var/run/cilium/hubble.sock
hubble-tls-cert-file                   /var/lib/cilium/tls/hubble/server.crt
hubble-tls-client-ca-files             /var/lib/cilium/tls/hubble/client-ca.crt
hubble-tls-key-file                    /var/lib/cilium/tls/hubble/server.key
identity-allocation-mode               crd
install-iptables-rules                 true
install-no-conntrack-iptables-rules    false
ipam                                   cluster-pool
kube-proxy-replacement                 disabled
monitor-aggregation                    medium
monitor-aggregation-flags              all
monitor-aggregation-interval           5s
node-port-bind-protection              true
nodes-gc-interval                      5m0s
operator-api-serve-addr                127.0.0.1:9234
preallocate-bpf-maps                   false
remove-cilium-node-taints              true
set-cilium-is-up-condition             true
sidecar-istio-proxy-image              cilium/istio_proxy
tunnel                                 vxlan
unmanaged-pod-watcher-interval         15
```

cilium status
![image](https://github.com/cilium/cilium/assets/65275144/c964fa9e-f38c-4f11-8d8a-46d6ba70dd57)

**Comment by dipankardas011** [comment]
Okay got it I have to shell in the cilium pod
Yes issue is still there
![image](https://github.com/cilium/cilium/assets/65275144/515e7f8d-025b-4b4a-8ba2-3658f7c7c384)

**Comment by dipankardas011** [comment]
> @joestringer Unable to run this command
> 
> Error message
> 
> ```shell
> $ cilium config view --all | grep -B 2 Mask
> unknown flag: --all
> ```
> 
> then tried
> 
> ```shell
> $ cilium config view
> agent-not-ready-taint-key              node.cilium.io/agent-not-ready
> annotate-k8s-node                      true
> arping-refresh-period                  30s
> auto-direct-node-routes                false
> bpf-lb-external-clusterip              false
> bpf-lb-map-max                         65536
> bpf-map-dynamic-size-ratio             0.0025
> bpf-policy-map-max                     16384
> cgroup-root                            /sys/fs/cgroup
> cilium-endpoint-gc-interval            5m0s
> cluster-id
> cluster-name                           default
> cluster-pool-ipv4-cidr                 10.0.0.0/8
> cluster-pool-ipv4-mask-size            24
> custom-cni-conf                        false
> debug                                  false
> disable-cnp-status-updates             true
> disable-endpoint-crd                   false
> enable-auto-protect-node-port-range    true
> enable-bandwidth-manager               false
> enable-bpf-clock-probe                 true
> enable-endpoint-health-checking        true
> enable-health-check-nodeport           true
> enable-health-checking                 true
> enable-host-legacy-routing             true
> enable-hubble                          true
> enable-ipv4                            true
> enable-ipv4-masquerade                 true
> enable-ipv6                            false
> enable-ipv6-masquerade                 true
> enable-k8s-terminating-endpoint        true
> enable-l2-neigh-discovery              true
> enable-l7-proxy                        true
> enable-local-redirect-policy           false
> enable-policy                          default
> enable-remote-node-identity            true
> enable-session-affinity                true
> enable-well-known-identities           false
> enable-xt-socket-fallback              true
> hubble-disable-tls                     false
> hubble-listen-address                  :4244
> hubble-socket-path                     /var/run/cilium/hubble.sock
> hubble-tls-cert-file                   /var/lib/cilium/tls/hubble/server.crt
> hubble-tls-client-ca-files             /var/lib/cilium/tls/hubble/client-ca.crt
> hubble-tls-key-file                    /var/lib/cilium/tls/hubble/server.key
> identity-allocation-mode               crd
> install-iptables-rules                 true
> install-no-conntrack-iptables-rules    false
> ipam                                   cluster-pool
> kube-proxy-replacement                 disabled
> monitor-aggregation                    medium
> monitor-aggregation-flags              all
> monitor-aggregation-interval           5s
> node-port-bind-protection              true
> nodes-gc-interval                      5m0s
> operator-api-serve-addr                127.0.0.1:9234
> preallocate-bpf-maps                   false
> remove-cilium-node-taints              true
> set-cilium-is-up-condition             true
> sidecar-istio-proxy-image              cilium/istio_proxy
> tunnel                                 vxlan
> unmanaged-pod-watcher-interval         15
> ```
> 
> cilium status ![image](https://user-images.githubusercontent.com/65275144/257593601-c964fa9e-f38c-4f11-8d8a-46d6ba70dd57.png)

Need one help can you, guide me on which exact file or folder or module to look for it?

**Comment by joestringer** [comment]
You can try using `git grep` or `git grep -i` for specific strings in order to find which files have references to those names. From there you can explore and tinker with the code.

**Comment by dipankardas011** [comment]
@joestringer will be refering to this PR #19141 and also git grep

**Comment by dipankardas011** [comment]
but still getting confused their is so much stuff

**Comment by dipankardas011** [comment]
@Nagendra-boyina can you tell how did you concluded which files were responsible and why?

**Comment by dipankardas011** [comment]
@joestringer I am unable to see the result of NAT46Prefix when running cilium in the kind cluster

**Comment by dipankardas011** [comment]
![image](https://github.com/cilium/cilium/assets/65275144/7f25e8f8-c44c-451c-869d-1c62dc881810)

**Comment by dipankardas011** [comment]
With the latest version it does not show

**Comment by dipankardas011** [comment]
![image](https://github.com/cilium/cilium/assets/65275144/9304109e-aad0-4da9-b6ff-41f59d44b932)
left is 1.11.6 and right is latest 1.14

**Comment by dipankardas011** [comment]
also you can try to check the headers which were set `/var/run/cilium/state/globals/node_config.h`

**Comment by dipankardas011** [comment]
![image](https://github.com/cilium/cilium/assets/65275144/9c9cc79d-b53d-4aef-8c0c-037e8cfd6387)

**Comment by dipankardas011** [comment]
I think this issue is not needed

**Comment by bewing** [comment]
It appears that this is a stale issue referencing removed code from commit 00b2318adb9765825fcc42ec156fd3919ad07d8a :
```
$ git log -SNAT46Prefix --
commit 00b2318adb9765825fcc42ec156fd3919ad07d8a
Author: Daniel Borkmann <daniel@iogearbox.net>
Date:   Fri Jan 28 16:18:29 2022 +0000

    cilium: Remove legacy agent-side NAT46 config
    
    Remove the old/legacy NAT46 endpoint related agent setting. It does not
    integrate with Kubernetes and was implemented back at a time where we also
    had the Docker plugin. There is little point in keeping it going forward as
    we plan to properly integrate NAT46/64 handling for services.
    
    Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
```

**Comment by joestringer** [comment]
Thanks for investigating @dipankardas011 @bewing ! Closing this one out.
